common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -201,7 +202,7 @@ class WebSocketRepositoryImpl @Inject constructor(
                             sendMessage(
                                 mapOf(
                                     "type" to "unsubscribe_events",
-                                    "subscription" to response.id
+                                    "subscription" to lastResponse!!.id
```

Comment:
`lastResponse` might be `null` at this point if a later subscription request returned `null`.

Reply:
but if a later subscription returned `null` wouldn't we already exit the method before reaching here?

Reply:
I don't think so? This is a callback, and returning if we receive `null` is outside the callback.

Imagine the first subscription works, receives ID `x` in `lastResponse`. The flow is setup and the callback is set. Now the app wants another subscription which fails, so `lastResponse` is set to `null` and no flow is created. When the app later stops listening to the first subscription, it will try to execute the callback but `lastResponse` is now `null` instead of something with ID `x`.

Reply:
hmm so in this particular case then my idea of keeping track of the last response is not good then 🤔

Reply:
I wouldn't consider this the 'last response', but rather the 'last subscription id'. It should be scoped to `eventType` like everything else in this function.

Reply:
so an additional variable(s) to keep track of the last ID per subscription when its not null? ok this is going to get out of hand if we add anything else lol. Am I understanding that correctly?

Reply:
I think

```
private var lastResponse = mutableMapOf<String, SocketResponse?>()
```

+ change `lastResponse` to `lastResponse[eventType]` when using it will work and isn't a lot of changes.

Reply:
Just saw your edit. Adding it now :)